lib/sqewer/local_connection.rb

```diff
@@ -0,0 +1,134 @@
+class Sqewer::LocalConnection < Sqewer::Connection
+  FAIL_AFTER_DELIVERIES = 10
+
+  def with_db(**k)
+    SQLite3::Database.open('sqewer-local-queue.sqlite3', **k) do |db|
```

Comment:
Why use this fixed filename instead of the one provided in SQS_QUEUE_URL? Also, it does not seem to be deleted anywhere, which may or may not be what you want. Finally, since this is mostly for tests the messages will often be discarded, so why not use the magic `:memory:` filename for sqlite and run the db in-memory?

Reply:
Also, you reopen the db for each call, even for reads? Why not have a persistent db connection?

Reply:
Because when you open a DB for writes you create a situation where it's very easy to get a SQLITE_BUSY. SQLite essentially puts a file lock on the DB you open for writing, but does so depending on the transaction grouping and/or the composition of the WAL. When you close the DB, you specifically release that flock, so that other processes/threads get a chance to step in. If you handle `receive_messages` within one DB block, for instance, you will get busy exceptions even in the single-threaded spec.

Reply:
`:memory:` is not used because the standard Sqewer setup is bi-process (some web worker produces jobs, some worker process consumes them). Even in development we are not using "inline" (in-process) workers, even though it is technically possible.